ingestion_server/ingestion_server/cleanup.py

```diff
@@ -272,24 +272,72 @@ def _clean_data_worker(rows, temp_table, sources_config, all_fields: list[str]):
     return cleaned_values
 
 
-def save_cleaned_data(result: dict) -> dict[str, int]:
-    log.info("Saving cleaned data...")
-    start_time = time.time()
+class CleanDataUploader:
+    # Number of records to buffer in memory before writing to disk
+    mem_buffer_size = 100
 
-    cleanup_counts = {field: len(items) for field, items in result.items()}
-    for field, cleaned_items in result.items():
-        # Skip the tag field because the file is too large and fills up the disk
-        if field == "tag":
-            continue
-        if cleaned_items:
-            with open(f"{field}.tsv", "a") as f:
-                csv_writer = csv.writer(f, delimiter="\t")
-                csv_writer.writerows(cleaned_items)
+    # Size (in MB) of local file buffer before writing to S3
+    disk_buffer_size = 10 * 1024 * 1024
 
-    end_time = time.time()
-    total_time = end_time - start_time
-    log.info(f"Finished saving cleaned data in {total_time},\n{cleanup_counts}")
-    return cleanup_counts
+    buffer = {
+        field: {"part": 1, "rows": []}
+        for field in _cleanup_config["tables"]["image"]["sources"]["*"]["fields"]
+    }
+
+    def __init__(self):
+        bucket_name = config("AWS_S3_BUCKET", default="openverse-catalog")
```

Comment:
As @obulat noted, while we use `openverse-catalog` in production, the default bucket locally is `openverse-storage`:

https://github.com/WordPress/openverse/blob/51fd23537b695114b203ace195041eaec0d7f8b4/catalog/env.template#L106-L105

Reply:
Gotcha, I added the existing bucket in production to `BUCKETS_TO_CREATE`. This way we have fewer differences between environments. I don't know why we would want to conserve `openverse-storage`. Let me know if there is a reason for this default bucket to be different between environments.

Reply:
I would prefer to replace the local `openverse-storage` with `openverse-catalog` here:
https://github.com/WordPress/openverse/blob/0ed4071577d846ff222342e5c296b723f5e7bb7e/catalog/env.template#L106
What do you think, @AetherUnbound ?
Otherwise, now locally we have 2 different buckets: one keeps the cleanup data and another - the data from provider scripts.

Reply:
If we do decide to use `openverse-catalog` everywhere locally, then it's best to fallback to the `OPENVERSE_BUCKET` env variable

Reply:
I changed it to use `OPENVERSE_BUCKET`!

Reply:
It looks like `openverse-catalog` is still being added to the `BUCKETS_TO_CREATE` list, shouldn't that be removed if we're deferring to `OPENVERSE_BUCKET`?

Reply:
@AetherUnbound `OPENVERSE_BUCKET` defaults to "openverse-catalog" in the Ingestion Server.